src/include/disks.h

```diff
@@ -143,6 +143,7 @@ struct disk {
   int mfs_idx;
   int part_image;               /* partition image */
   int genmbr;
+  int isfat32;                  /* If the image contains a fat32 filesystem */
```

Comment:
That's weird.
Maybe instead reserve a buffer
for vbr here, and access it when
needed?
But still weird as I don't see how
does this work with multi-partition
images, where you have more than
1 vbr or more than one `isfat32`.
This all makes me wonder if this is
a right thing to do.
How about splitting a patch set that
dials with partition image w/o MBR,
into a separate patch-set? That one
would be clear. The rest should have
to be separately considered.

Reply:
> That's weird.

Indeed, I'm not that keen on it myself.

> Maybe instead reserve a buffer
for vbr here, and access it when
needed?

Can do, obviously it will only be populated when we have a VBR image.

> But still weird as I don't see how
does this work with multi-partition
images, where you have more than
1 vbr or more than one isfat32.

It isn't used with anything other than VBR images. If you have an MBR image containing multiple partitions/VBRs then it isn't set or used. It's used purely when constructing an MBR.

> How about splitting a patch set that
dials with partition image w/o MBR,
into a separate patch-set?

Yes it's getting a little out of hand.

Reply:
Switched to your method of changing the type between auto and setup functions, so complexity has now gone.